### DISCUSSION TITLE
AttributeError: 'array' object has no attribute 'astype'

### DISCUSSION THREAD

**Discussion by peterchencyc** [NVIDIA/warp#55] [General]
`File "/opt/conda/lib/python3.8/site-packages/warp/sim/integrator_euler.py", line 1881, in simulate`
`    x = state_out.particle_qd.astype(dtype=float)`
`AttributeError: 'array' object has no attribute 'astype'`

**Comment by mmacklin** [comment]
Hi @peterchencyc, yes, the `VariationalImplicitIntegrator` is currently under heavy construction and not actively tested. It's likely we'll replace this class in the near future with a separate backward Euler integrator. I will make a task to remove this in the next version.

Cheers,
Miles

**Comment by Changy1506** [comment]
Hi, I think the `astype` here is converting the data type of the array, from `warp.vec3` to `float`. To run this part, I created another array of float with 3 * state_out.particle_qd.size, and wrote a simple kernel function to initialize the new array with the data from `state_out.particle_qd`.
